src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java

```diff
@@ -243,18 +244,11 @@ public File createArchive()
             }
         }
 
+        final String archiverName = containsModuleDescriptor ? "mjar" : "jar";
```

Comment:
why final? no real need of this

Reply:
oh oh, I know people they insinst on final...